client/post-editor/media-modal/index.jsx

```diff
@@ -107,40 +105,10 @@ export const EditorMediaModal = React.createClass( {
 	},
 
 	confirmSelection: function() {
-		var selectedItems = this.props.mediaLibrarySelectedItems,
-			gallerySettings = this.state.gallerySettings,
-			media, stat;
-		const site = this.props.site;
-
-		if ( ! this.props.visible ) {
-			return;
-		}
-
-		if ( ModalViews.GALLERY === this.props.view ) {
-			if ( gallerySettings && 'individual' === gallerySettings.type ) {
-				media = gallerySettings.items.map( item => markup.get( site, item ) ).join( '' );
-			} else {
-				media = MediaUtils.generateGalleryShortcode( gallerySettings );
-			}
-			stat = 'insert_gallery';
-		} else {
-			media = selectedItems.map( item => markup.get( site, item ) ).join( '' );
-			stat = 'insert_item';
-		}
-
-		if ( some( selectedItems, 'transient' ) ) {
-			PostActions.blockSave( 'MEDIA_MODAL_TRANSIENT_INSERT' );
-		}
-
-		if ( media ) {
-			this.props.onInsertMedia( media );
-
-			if ( stat ) {
-				analytics.mc.bumpStat( 'editor_media_actions', stat );
-			}
-		}
+		const { mediaLibrarySelectedItems } = this.props;
+		const { gallerySettings } = this.state;
 
-		this.props.onClose( selectedItems );
+		this.props.onClose( mediaLibrarySelectedItems, { gallerySettings } );
```

Comment:
Should we normalize the output format here. By this, I mean, maybe we should have a documented output that could look like this: 

```js 
{ type: 'gallery', items: [], settings: {} }
```

and 

```js
{ type: 'media', item: 1 }
```

This would ease switching on the two possible outputs of the `MediaModal` ?